eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -33,36 +33,34 @@ jobs:
           build.args: ' -DTRANSPORT_PAHO=ON'
           test_type: 'iot'
           os: 'macos'
-        # Disable live testing for these matrix entries until we have something to
-        # test here.
-        # Linux_x64_with_samples:
-        #   Pool: azsdk-pool-mms-ubuntu-1804-general
-        #   OSVmImage: MMSUbuntu18.04
-        #   vcpkg.deps: 'curl[ssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        #   build.args: ' -DTRANSPO.RT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
-        # Win_x86_with_sampldes:
-        #   Pool: azsdk-pool-mms-win-2019-general
-        #   OSVmImage: MMS2019
-        #   vcpkg.deps: 'curl[winssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
-        #   CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        #   CMAKE_GENERATOR_PLATFORM: Win32
-        #   build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=WIN32'
-        # Win_x64_with_samples:
-        #   Pool: azsdk-pool-mms-win-2019-general
-        #   OSVmImage: MMS2019
-        #   vcpkg.deps: 'curl[winssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
-        #   CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        #   CMAKE_GENERATOR_PLATFORM: x64
-        #   build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=WIN32'
-        # MacOS_x64_with_samples:
-        #   Pool: Azure Pipelines
-        #   OSVmImage: 'macOS-10.15'
-        #   vcpkg.deps: 'curl[ssl]'
-        #   VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-        #   build.args: ' -DTRANSPORT_CURL=ON  -DAZ_PLATFORM_IMPL=POSIX'
+        Linux_x64_with_samples:
+          Pool: azsdk-pool-mms-ubuntu-1804-general
+          OSVmImage: MMSUbuntu18.04
+          vcpkg.deps: 'curl[ssl]'
+          VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
+        Win_x86_with_samples:
+          Pool: azsdk-pool-mms-win-2019-general
+          OSVmImage: MMS2019
+          vcpkg.deps: 'curl[winssl]'
+          VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
+          CMAKE_GENERATOR: 'Visual Studio 16 2019'
+          CMAKE_GENERATOR_PLATFORM: Win32
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=WIN32'
+        Win_x64_with_samples:
+          Pool: azsdk-pool-mms-win-2019-general
+          OSVmImage: MMS2019
+          vcpkg.deps: 'curl[winssl]'
+          VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
+          CMAKE_GENERATOR: 'Visual Studio 16 2019'
+          CMAKE_GENERATOR_PLATFORM: x64
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=WIN32'
+        MacOS_x64_with_samples:
+          Pool: Azure Pipelines
+          OSVmImage: 'macOS-10.15'
+          vcpkg.deps: 'curl[ssl]'
+          VCPKG_DEFAULT_TRIPLET: 'x64-osx'
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
```

Comment:
We need `os: 'macos'`


https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1187179&view=logs&j=d1a859cc-2145-5e72-055c-d2e5fdad58d8&t=08df4c10-81fa-580a-f797-71fb278bea08

> Expanded: and(contains(' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX', '-DTRANSPORT_CURL=ON'), or(eq(Null, 'linux'), eq(Null, 'macos')))
Result: False